mcp/tools.go

```diff
@@ -1076,3 +1176,151 @@ func WithBooleanItems(opts ...PropertyOption) PropertyOption {
 		schema["items"] = itemSchema
 	}
 }
+
+//
+// Output Schema Configuration Functions
+//
+
+// WithOutputSchema sets the output schema for the Tool.
+// This allows the tool to define the structure of its return data.
+func WithOutputSchema(schema ToolSchema) ToolOption {
+	return func(t *Tool) {
+		t.OutputSchema = schema
+	}
+}
+
+// WithOutputBoolean adds a boolean property to the tool's output schema.
+// It accepts property options to configure the boolean property's behavior and constraints.
+func WithOutputBoolean(name string, opts ...PropertyOption) ToolOption {
+	return func(t *Tool) {
+		// Initialize output schema if not set
+		if t.OutputSchema.Type == "" {
+			t.OutputSchema.Type = "object"
+		}
+
+		schema := map[string]any{
+			"type": "boolean",
+		}
+
+		for _, opt := range opts {
+			opt(schema)
+		}
+
+		// Remove required from property schema and add to OutputSchema.required
+		if required, ok := schema["required"].(bool); ok && required {
+			delete(schema, "required")
+			t.OutputSchema.Required = append(t.OutputSchema.Required, name)
+		}
+
+		t.OutputSchema.Properties[name] = schema
+	}
+}
+
+// WithOutputNumber adds a number property to the tool's output schema.
+// It accepts property options to configure the number property's behavior and constraints.
+func WithOutputNumber(name string, opts ...PropertyOption) ToolOption {
+	return func(t *Tool) {
+		// Initialize output schema if not set
+		if t.OutputSchema.Type == "" {
+			t.OutputSchema.Type = "object"
+		}
+
+		schema := map[string]any{
+			"type": "number",
+		}
+
+		for _, opt := range opts {
+			opt(schema)
+		}
+
+		// Remove required from property schema and add to OutputSchema.required
+		if required, ok := schema["required"].(bool); ok && required {
+			delete(schema, "required")
+			t.OutputSchema.Required = append(t.OutputSchema.Required, name)
+		}
+
+		t.OutputSchema.Properties[name] = schema
+	}
+}
```

Comment:
Do we need a separate schema generation function for each datatype because of backwards compatibility? Like can we reuse the same methods we use for input schema?

Reply:
Got it, I understand. In future discussions, I’ll respond directly based on my own understanding. I reflected on it just now and realized that the AI did generate a lot of unnecessary extra information.